src/DataSource/DoctrineCollectionDataSource.php

```diff
@@ -0,0 +1,272 @@
+<?php
+
+/**
+ * @copyright   Copyright (c) 2015 ublaboo <ublaboo@paveljanda.com>
+ * @author      Jakub Kontra <me@jakubkontra.cz>
+ * @author      Pavel Janda <me@paveljanda.com>
+ * @package     Ublaboo
+ */
+
+namespace Ublaboo\DataGrid\DataSource;
+
+use Doctrine\Common\Collections\Collection;
+use Doctrine\Common\Collections\Criteria;
+use Ublaboo\DataGrid\Filter;
+use Ublaboo\DataGrid\Utils\Sorting;
+
+class DoctrineCollectionDataSource extends FilterableDataSource implements IDataSource
+{
+
+	/**
+	 * @var Collection
+	 */
+	protected $data_source;
+
+	/**
+	 * @var string
+	 */
+	protected $primary_key;
+
+	/**
+	 * @var Criteria
+	 */
+	protected $criteria;
+
+	/**
+	 * @var Collection
+	 */
+	private $matched;
+
+
+	/**
+	 * @param Collection  $collection
+	 * @param string      $primary_key
+	 */
+	public function __construct(Collection $collection, $primary_key)
+	{
+		$this->criteria = Criteria::create();
+		$this->data_source = $collection;
+		$this->primary_key = $primary_key;
+	}
+
+
+	protected function initialize()
```

Comment:
Why are you using that method with return statement, when the return value is not used?

Also, can you please rename the method? I just don't like so much these magic names. `initialize` does not tell much about itself..